src/services/files.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * name : files.js
+ * author : Aman Gupta
+ * created-date : 03-Nov-2021
+ * Description : files helper.
+ */
+
+// Dependencies
+const cloudServices = require('@generics/cloud-services')
+const httpStatusCode = require('@generics/http-status')
+const common = require('@constants/common')
+const utils = require('@generics/utils')
+const responses = require('@helpers/responses')
+const { cloudClient } = require('@configs/cloud-service')
+const endpoints = require('@constants/endpoints')
+const cloudStorage = process.env.CLOUD_STORAGE_PROVIDER
+const bucketName = process.env.CLOUD_STORAGE_BUCKETNAME
+
+module.exports = class FilesHelper {
+	/**
+	 * Get Signed Url
+	 * @method
+	 * @name getSignedUrl
+	 * @param {JSON} payloadData  request body.
+	 * @param {string} referenceType- what type of document
+	 * @param {string} userId  -  userId
+	 * @param {boolean} serviceUpload - needed for nic server
+	 * @returns {JSON} - Response contains signed url
+	 */
+	static async getSignedUrl(payloadData, referenceType, userId = '', serviceUpload = false) {
+		try {
+			let payloadIds = Object.keys(payloadData)
+
+			let result = {
+				[payloadIds[0]]: {},
+			}
+
+			let folderPath = ''
+
+			if (referenceType == common.CERTIFICATE) {
+				folderPath =
+					common.CERTIFICATE_PATH + userId + '/' + payloadIds[0] + '/' + utils.generateUniqueId() + '/'
```

Comment:
@ankit1shahu why do we need to store in the different paths? If we delete user data we need to search in all folder and delete